dot/network/notifications.go

```diff
@@ -226,26 +235,41 @@ func (s *Service) sendData(peer peer.ID, hs Handshake, info *notificationsProtoc
 			return
 		}
 
-		hs, err := s.readHandshake(stream, decodeBlockAnnounceHandshake)
-		if err != nil {
-			logger.Trace("failed to read handshake", "protocol", info.protocolID, "peer", peer, "error", err)
+		var hs Handshake
+		hstimeout, cancel := context.WithTimeout(s.ctx, handshakeTimeout)
+		defer cancel()
+
+		select {
+		case <-hstimeout.Done():
```

Comment:
this might slow down the performance of the node, since creating a timeout requires `malloc`ing and doing this for every outbound stream is potentially too often. I'll profile it and let you know

Reply:
Maybe a Ticker could be better in terms of performance?